### ISSUE TITLE
error with getcwd() on MooseFS 3.0.112-1

### ISSUE THREAD

**Issue by tamas-ac** [moosefs/moosefs#357]
BUG Report: error with getcwd() on MooseFS 3.0.112-1

Platform: MooseFS running in a sidecar container (Debian/buster base image) on Kubernetes (CentOS on the nodes)

We've recently updated our MooseFS client from 3.0.111-1 to 3.0.112-1 in our sidecar container from the official repository, and noticed something strange: in the main pod container, getcwd returned errors like this:

`sh: 0: getcwd() failed: No such file or directory`

Listing the directory and creating new files in it worked in the terminal though.

In this container, MooseFS client is not installed, the MooseFS folders are shared via volume mounts. We never had such problems with 3.0.111-1. Now we've reverted the upgrade as this is a blocker for us.

Can this be related to the dentry invalidator change in 3.0.112-1?

**Comment by chogata** [comment]
Yes, it could. We would have to test it, but I suspect MooseFS might not be able to determine which directories are currently CWDs inside the containers.

We will run tests and try to find a working solution.

**Comment by SlackCrow** [comment]
We are using the latest version of MooseFS(3.0.116-1) and having a similar problem. We have a Ubuntu container running on Kubernetes. In this container, MooseFS client is not installed; a MooseFS folder is shared via volume mounts.

Just like what tarmas-ac experienced, listing the directory and creating new files in it worked in the terminal. However, when we try to import PyTorch(python -c "import torch") or run a script that import PyTorch, we get an error like the one below. 
`Intel MKL FATAL ERROR: Cannot load /opt/conda/lib/libmkl_core.so.`

We were confused at first and thought we were running a wrong version of Python installed or something, so we ran which python and got an error like the one below.
`shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory
/opt/conda/bin/python`

If we cd to a folder that is under the mounted folder, it works just fine. For example, if the folder mounted is src-shared-mfs, Python scripts that are in src-shared-mfs don't work, but Python scripts that are in src-shared-mfs/python work.

What perplexes us the most is that this doesn't happen all the time. At first when we start the pod it works fine, but then if we cd to another folder and back it stops working usually.

**Comment by chogata** [comment]
Thank you for the report, we will investigate.

**Comment by xandrus** [comment]
Hi @SlackCrow,

Can you tell us more about the K8s?
Basically, we would like to know which version of K8s you are using and what your underlying operating system is.

**Comment by rafamiga** [comment]
@SlackCrow You may take look at the issue I had with k8s – it might be related to your problem.

**Comment by SlackCrow** [comment]
> @SlackCrow You may take look at the issue I had with k8s – it might be related to your problem.
> 
> #339

It might be related but with the older version of the MooseFS client it seems to work fine, so I am not so sure.

**Comment by chogata** [comment]
Hi @SlackCrow 

I see that the default kernel for Ubuntu 18.04 is 4.15. Are you using that? If yes, you need to upgrade to at least 4.19, otherwise you will have this problem. The dentry invalidator, which I'm 99% sure is your problem (this has to do with mounting in containers), needs to be active in kernel 4.15, otherwise you will get EBUSY error problems. So basically the best solution is just upgrade to a newer kernel version.

**Comment by SlackCrow** [comment]
> Hi @SlackCrow
> 
> I see that the default kernel for Ubuntu 18.04 is 4.15. Are you using that? If yes, you need to upgrade to at least 4.19, otherwise you will have this problem. The dentry invalidator, which I'm 99% sure is your problem (this has to do with mounting in containers), needs to be active in kernel 4.15, otherwise you will get EBUSY error problems. So basically the best solution is just upgrade to a newer kernel version.

D'oh! Totally missed the >=4.19 part. We upgraded the kernel to 5.0.0-15 and everything seems to work fine now. We decided to upgrade some of our testing servers to 5.0.0-15 and stick with 3.0.111-1 for the rest for now. 
Thank you for the pointer! Greatly appreciated.
